### PR TITLE
Feature/extend eventform

### PR DESCRIPTION
**Pull request by Sagensagen**
# :gear:  Event editing ++
Fixes a bug  in response handler to cooperate with JSON structure types.
Adds margin bottom to event statistics for better looks :lipstick: 
Joining an event will show only the fields that are applicable for the respective event.(Food allergies, transport, price)
Adds to editEvent:
- Food
- Transportation
- Date+time(dateTime)
- price

### Todo(?)
Not all fields are in updateEvent are validated, but implicitly a correct value. Should maybe add?

### CODE REVIEW

File: src/components/molecules/event/eventBody/EventBody.tsx [hunk 1]
```diff
@@ -13,40 +13,49 @@ import {
...
+    event.transportation ?? false
+  );
   const initalValue = {
     title: event.title,
     description: event.description,
-    date: event.date,
+    date: event.date.split('T')[0], //sets [YYYY-MM-DD]
+    time: timeDate.getHours() + ':' + timeDate.getMinutes(), //sets [HH:MM] and not [HH:MM:SS]
     address: event.address,
+    price: event.price.toString(),
```
Comment (FredrikMorstad) [inline src/components/molecules/event/eventBody/EventBody.tsx:44]: Is there any reason for converting the price to string?

**Review by FredrikMorstad — CHANGES_REQUESTED** [review]
Code looks good, however the commit for the request fix should say that it fixed bug related to handling empty response.

File: src/components/molecules/event/eventBody/EventBody.tsx (hunk 1 above)
Comment (Sagensagen) [reply src/components/molecules/event/eventBody/EventBody.tsx:44]: `export const priceValidator = (price: string) => {
  return price.length ? undefined : ['Arrangement pris må fylles ut'];
};`
Pricevalidator validates price as `String`

**Review by FredrikMorstad — APPROVED** [review]
LGTM :rocket:

**Pull request merged**